File: crawler.py
import requests as req
from bs4 import BeautifulSoup as bs

#python3로 멜로차트 TOP100의 순위, 곡정보, 앨범, 좋아요수를 크롤링

def crawl_melon_chart():
    url = 'https://www.melon.com/chart/index.htm'

    # HTTP GET 요청
    response = req.get(url)
    html = response.text

    # BeautifulSoup 객체 생성
    soup = bs(html, 'html.parser')

    # 멜론 차트 데이터 추출
    chart_list = []

    chart_elements = soup.select('.lst50, .lst100')
    for element in chart_elements:
        rank = element.select_one('.rank').text.strip()
        title = element.select_one('.ellipsis.rank01 > span > a').text.strip()
        artist = element.select_one('.ellipsis.rank02 > a').text.strip()
        album = element.select_one('.ellipsis.rank03 > a').text.strip()
        likes = element.select_one('.button_etc.like > span.cnt').text.strip()

        song_info = {
            'rank': rank,
            'title': title,
            'artist': artist,
            'album': album,
            'likes': likes
        }

        chart_list.append(song_info)

    # 추출한 멜론 차트 데이터 출력
    for song in chart_list:
        print(f"Rank {song['rank']}: {song['title']} - {song['artist']}")
        print(f"Album: {song['album']}")
        print(f"Likes: {song['likes']}")
        print("-------------------------")

# 멜론 차트 크롤링 함수 호출
crawl_melon_chart()

File: crawler.js
const axios = require('axios'); //npm packages 설치 필요
const cheerio = require('cheerio'); //cheerio -> npm -i cheerio 설치 필요

// 멜론 차트 크롤링 함수
async function crawlMelonChart() {
  try {
    const url = 'https://www.melon.com/chart/index.htm'; // 멜론 차트 URL

    // 웹 페이지 HTML 가져오기
    const response = await axios.get(url);
    const html = response.data;

    // Cheerio 라이브러리를 사용하여 HTML 파싱
    const $ = cheerio.load(html);

    // 멜론 차트 데이터 추출
    const chartList = [];

    //100위 링크
    $('.lst50, .lst100').each((index, element) => {
      const rank = $(element).find('.rank').text().trim();
      const title = $(element).find('.ellipsis.rank01 > span > a').text().trim(); 
      const artist = $(element).find('.ellipsis.rank02 > a').text().trim();
      const album = $(element).find('.ellipsis.rank03 > a').text().trim();
      const likes = $(element).find('.button_etc.like > span.cnt').text().trim();

      const songInfo = {
        rank,
        title,
        artist,
        album,
        likes,
      };

      chartList.push(songInfo);
    });

    // 추출한 멜론 차트 데이터 출력
    chartList.forEach((song) => {
      console.log(`Rank ${song.rank}: ${song.title} - ${song.artist}`);
      console.log(`Album: ${song.album}`);
      console.log(`Likes: ${song.likes}`);
      console.log('-------------------------');
    });
  } catch (error) {
    console.error('Error occurred during crawling:', error);
  }
}

// 멜론 차트 크롤링 함수 호출
crawlMelonChart();
